packages/datadog-plugin-mongodb-core/src/index.js

```diff
@@ -1,6 +1,115 @@
 'use strict'
 
-const unified = require('./unified')
-const legacy = require('./legacy')
+const Plugin = require('../../dd-trace/src/plugins/plugin')
+const { storage } = require('../../datadog-core')
+const analyticsSampler = require('../../dd-trace/src/analytics_sampler')
 
-module.exports = [].concat(unified, legacy)
+const COMMANDS = ['command', 'insert', 'update', 'remove', 'getMore', 'query', '_find', 'killCursors']
+class MongodbCorePlugin extends Plugin {
+  static get name () {
+    return 'mongodb-core'
+  }
+
+  addSubs (func, start) {
+    this.addSub(`apm:mongodb:${func}:start`, start)
+    this.addSub(`apm:mongodb:${func}:end`, this.exit.bind(this))
+    this.addSub(`apm:mongodb:${func}:error`, errorHandler)
+    this.addSub(`apm:mongodb:${func}:async-end`, defaultAsyncEnd)
+  }
+
+  startSpan (customTags, store, ns, cmd, topology, operationName) {
+    const query = getQuery(cmd)
+    const resource = getResource(ns, query, operationName)
+    const tags = {
+      'service.name': this.config.service || `${this.tracer._service}-mongodb`,
+      'resource.name': resource,
+      'span.type': 'mongodb',
+      'span.kind': 'client',
+      'db.name': ns
+    }
+
+    for (const tag in customTags) {
+      tags[tag] = customTags[tag]
+    }
+    const span = this.tracer.startSpan('mongodb.query', {
+      childOf: store ? store.span : null,
+      tags
+    })
+
+    if (query) {
+      span.setTag('mongodb.query', query)
+    }
+
+    const options = topology && topology.s && topology.s.options
```

Comment:
This should not be needed on the subscriber side as this should be normalized on the publisher side.

Reply:
Same for the below code.